akvo/rsr/filters.py

```diff
@@ -98,26 +98,45 @@ def get_locations(location, locations):
 def location_choices(qs):
     """From a queryset get possible location filter choices"""
 
-    if qs.model is Project:
-        location_model = ProjectLocation
-    elif qs.model is ProjectUpdate:
+    country_ids = (
+        get_recipient_country_ids(qs) if qs.model is Project else
```

Comment:
Sure the filter doesn't show all project but I think that if a project is associated with a country it should be possible to find. And by extension I'd say that if there's no recipient country but we have a country from project location we should display that.